### PR TITLE
[FLOC-4282] publish-installer-images now requires an AMI map input

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-4282

Stacked on #2687

This allows me to feed the output of `./admin/ami-search-ubuntu` into `./admin/publish-installer-images`.

See #2672 for proposed documentation.

### CODE REVIEW

File: admin/installer/_images.py [hunk 1]
```diff
@@ -163,14 +147,14 @@ class PackerConfigure(PClass):
...
-    :ivar source_ami: The AMI ID to use as the base image.
+    :ivar source_ami_map: The AMI map containing base images.
     """
     build_region = field(type=RegionConstant, mandatory=True)
     publish_regions = pvector_field(item_type=RegionConstant)
     template = field(type=unicode, mandatory=True)
     distribution = field(type=unicode, mandatory=True)
     configuration_directory = field(type=FilePath, initial=PACKER_TEMPLATE_DIR)
-    source_ami = field(type=unicode, mandatory=True)
+    source_ami_map = pmap_field(key_type=RegionConstant, value_type=unicode)
```
Comment (myechuri) [inline admin/installer/_images.py]: Should this have ``mandatory=True``? Thanks.

Comment (wallrj) [inline admin/installer/_images.py]: No. Confusingly the `*_field` functions produce mandatory fields by default.
 * https://pyrsistent.readthedocs.org/en/latest/api.html#pyrsistent.pmap_field

Comment (myechuri) [inline admin/installer/_images.py]: Ah ok, thanks!

**Comment by myechuri** [comment]
Looks look, thanks, @wallrj .

**Comment by wallrj** [comment]
It seems impossible to get the cinder driver functional tests to pass right now, so I'll ignore that failure and merge.

```
00:55:37 2016-03-09 00:55:29+0000 [-] ELIOT Extracted Traceback:
00:55:37 	Traceback (most recent call last):
00:55:37 	  File "/tmp/workspace/ClusterHQ-flocker/ami-map-input-FLOC-4282/run_trial_for_cinder_storage_driver_on_Ubuntu_trusty_flocker.node.agents.functional/flocker/node/agents/test/test_blockdevice.py", line 356, in detach_destroy_volumes
00:55:37 	  File "/tmp/workspace/ClusterHQ-flocker/ami-map-input-FLOC-4282/run_trial_for_cinder_storage_driver_on_Ubuntu_trusty_flocker.node.agents.functional/flocker/node/agents/cinder.py", line 562, in detach_volume
00:55:37 	  File "/tmp/workspace/ClusterHQ-flocker/ami-map-input-FLOC-4282/run_trial_for_cinder_storage_driver_on_Ubuntu_trusty_flocker.node.agents.functional/flocker/node/agents/cinder.py", line 359, in wait_for_volume_state
00:55:37 	  File "/tmp/workspace/ClusterHQ-flocker/ami-map-input-FLOC-4282/run_trial_for_cinder_storage_driver_on_Ubuntu_trusty_flocker.node.agents.functional/flocker/common/_retry.py", line 233, in poll_until
00:55:37 	  File "/tmp/workspace/ClusterHQ-flocker/ami-map-input-FLOC-4282/run_trial_for_cinder_storage_driver_on_Ubuntu_trusty_flocker.node.agents.functional/flocker/node/agents/cinder.py", line 336, in reached_desired_state
00:55:37 	UnexpectedStateException: Unexpected state while waiting for volume. Expected Volume: <Volume: 3d6d196c-9b6b-44b2-a5bd-5985a77e5fb2>, Expected State: u'available', Reached State: u'in-use'
00:55:37 	
```

**Pull request merged**